### PR TITLE
CLDR-11585 mavenize (2020 parallel version)

### PR DESCRIPTION
**Pull request by srl295**
CLDR-11585

So, it turns out this can be done a little bit in parallel after all.

- add pom.xml and use the _current_ directory structure. Once this is working, then switch over.
- no tests yet (this needs code refactoring)
- cldr-apps builds and runs OK
- it's possible some of the libs are slightly out of date, but that's easily fixable.
- using a temporary repo for ICU4J/utilities until https://github.com/unicode-org/icu/pull/1270 gets merged.

### CODE REVIEW

**Comment by srl295** [comment]
blocked on https://github.com/unicode-org/icu/pull/1270

**Comment by srl295** [comment]
So, this isn't great, and does not include icu4j libs from #702 yet.

But it's a step forward. Can we get this merged, with a parallel, non-blocking maven build?

File: tools/cldr-apps/pom.xml
Comment (hagbard) [inline tools/cldr-apps/pom.xml:173]: Obviously build file indentation etc.
And it would be nice if any commented out dependencies were either comment as to why, to removed.

File: tools/cldr-apps/pom.xml
Comment (hagbard) [inline tools/cldr-apps/pom.xml:206]: Can you reference a CLDR ticket number in the TODO ?

File: tools/cldr-apps/pom.xml
Comment (hagbard) [inline tools/cldr-apps/pom.xml:246]: "the pom.xml directory" --> "the directory this POM file is in" ?

File: tools/java/pom.xml
Comment (hagbard) [inline tools/java/pom.xml:92]: Can this be ordered alphabetically ? It nearly is already.

File: tools/java/pom.xml
Comment (hagbard) [inline tools/java/pom.xml:127]: It sounds like there's a vaguely structured plan behind some of these TODOs. Having a CLDR ticket for the discussion would be great.

File: tools/java/pom.xml
Comment (hagbard) [inline tools/java/pom.xml:156]: I assume you can auto-indent everything easily enough before merging.

File: .github/workflows/mvn-settings.xml
Comment (hagbard) [inline .github/workflows/mvn-settings.xml:1]: Copyright?
Meta-comment on the purpose of these files?

File: .github/workflows/mvn-settings.xml
Comment (hagbard) [inline .github/workflows/mvn-settings.xml:48]: Newline at eof?

**Review by hagbard — COMMENTED** [review]
Just a few quick comments.

File: tools/pom.xml
Comment (echeran) [inline tools/pom.xml:16]: Since this is the parent (root?) pom.xml, this is where you can "refactor" (de-duplicate) specifying the version numbers of commonly-used dependencies and plugins via [`dependencyManagement`](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management).

Similarly, using `<dependencyManagement>` in `tools/cldr-apps/pom.xml` probably isn't doing much for you, as it is a child/leaf pom.xml.

Also, if you refactor `./java/pom.xml` and `./cldr-apps/pom.xml` to de-duplicate using the parent-child inheritance, then you can run one of the "child" pom.xmls (ex: `./java/pom.xml`) with the proper inerhitance/interpolation in effect by running this command in the directory of the root pom.xml:  `mvn -am -pl java/pom.xml`.

I think the only catch with Maven is that the version number of artifact that the pom.xml defines can't be inherited, and it must be hard-coded.

I don't know if this is useful, but I know that there are [Maven plugins](http://maven.apache.org/maven-release/maven-release-plugin/index.html) out there to manage the incrementing of versions from N to (N+1)-SNAPSHOT to N+1, etc.  They can can manage the version value in the submodule pom.xml's as well.

File: tools/cldr-apps/pom.xml
Comment (srl295) [reply tools/cldr-apps/pom.xml:173]: I'll run everything through Eclipse's cleanup and fix the dependency docs, thanks.

File: tools/cldr-apps/pom.xml
Comment (srl295) [reply tools/cldr-apps/pom.xml]: It's this same one, CLDR-11585. To solve this I need to rearrange the source files which breaks everyone. That's why I want To do this in stages.

File: tools/java/pom.xml
Comment (srl295) [reply tools/java/pom.xml]: I will update CLDR-11585

File: tools/pom.xml
Comment (srl295) [reply tools/pom.xml:16]: > Similarly, using <dependencyManagement> in tools/cldr-apps/pom.xml probably isn't doing much for you, as it is a child/leaf pom.xml.

This turned out to help the build be able to find it as a sibling dependency.

Will do the rest, I quickly scrambled my workspace trying to do this refactoring. 

I'd really like to rename 'java' to something, perhaps 'cldr4j' or 'cldr-tools'. Otherwise it's misnamed.

Comment (srl295) [reply tools/pom.xml:16]: corrected… push coming

File: tools/cldr-apps/pom.xml
Comment (srl295) [reply tools/cldr-apps/pom.xml]: I ended up removing/replacing this comment. It's more important to note that the point of this is to temporarily avoid renaming all of the source files.

File: tools/cldr-apps/pom.xml
Comment (echeran) [inline tools/cldr-apps/pom.xml:151]: Now that you have `<dependencyManagement>` and `<pluginManagement>` in an ancestor pom.xml with duplicated version info specified there, is it now possible here to remove the tags `<pluginManagement>`, `</pluginManagement>`, and elements of the form `<version>...</version>` (leaving every other tag nested inside them as-is)?  This should not change the behavior in any way, of course.

File: tools/cldr-apps/pom.xml
Comment (echeran) [inline tools/cldr-apps/pom.xml:220]: I didn't remember / realize this, but pluginManagement and dependencyManagement propagate the configuration and other information for their respective plugin and dependency declarations, not just the version numbers.  These configuration details are duplicated in the root pom.xml, which `tools/java/pom.xml` is inheriting.  You could remove these details here, similarly, to clean up and allow inheritance.  Of course, verify that what I said is correct and the behavior is unchanged.

Or, alternatively, if you prefer, you could not use the inheritance mechanism for the configuration details, leaving it to be explicitly configured only in leaf pom.xml files and not in the root pom.xml.

I don't have a strong opinion on that, either way.

File: tools/cldr-apps/pom.xml
Comment (srl295) [reply tools/cldr-apps/pom.xml:151]: Yes, I will make sure the *Management tags are rolled up into the parent.

File: tools/cldr-apps/pom.xml
Comment (srl295) [reply tools/cldr-apps/pom.xml:220]: @echeran
- I've moved *all* `*Management` into the parent pom
- the only common configuration that makes sense to make common (and I have) is the githash stuff, everything else is per-project.

**Review by echeran — APPROVED** [review]
LGTM

**Comment by srl295** [comment]
@macchiati @yumaoka any thoughts on this?

**Comment by macchiati** [comment]
I'm not particularly up on maven, so I would go by what Elango says. I'm glad to rubberstamp unless Yoshito has time to review.

**Review by macchiati — APPROVED** [review]

**Pull request merged**
